google-home-community.groovy

```diff
@@ -2175,20 +2204,34 @@ private executeCommand_RotateAbsolute(deviceInfo, command) {
 
 @SuppressWarnings('UnusedPrivateMethod')
 private executeCommand_SetFanSpeed(deviceInfo, command) {
-    checkMfa(deviceInfo.deviceType, "Set Fan Speed", command)
-    def fanSpeedTrait = deviceInfo.deviceType.traits.FanSpeed
-    def fanSpeed = command.params.fanSpeed
-
-    deviceInfo.device."${fanSpeedTrait.setFanSpeedCommand}"(fanSpeed)
-    return [
-        [
-            (fanSpeedTrait.currentSpeedAttribute): fanSpeed,
-        ],
-        [
-            currentFanSpeedSetting: fanSpeed,
-        ],
-    ]
-}
+	checkMfa(deviceInfo.deviceType, "Set Fan Speed", command)
+	def fanSpeedTrait = deviceInfo.deviceType.traits.FanSpeed
+
+	if (fanSpeedTrait.supportsFanSpeedPercent && command.params.fanSpeedPercent) {
+		def fanSpeedpercent = command.params.fanSpeedPercent
+
+		deviceInfo.device."${fanSpeedTrait.setFanSpeedPercentCommand}"(fanSpeedpercent)
+		return [
+			[
+				(fanSpeedTrait.currentFanSpeedPercent): fanSpeedpercent,
+			],
+			[
+				currentFanSpeedPercent: fanSpeedpercent,
```

Comment:
It probably doesn't matter too much (Google only uses these values to generate voice responses), but I think both paths should return both the speed setting and (if supported) the speed percent.

Reply:
If fixed those typos you caught and now it works perfectly.

Previously, on the Google Hub (full size, not a hub mini), it would show a slider and you could change the percentage (and Hubitat would show the change),  but a few seconds later, the Hub on the slider slid back to 0 (though Hubitat didn't change).

Now everything works as expected.  So, I'm not sure what you meant by "I think both paths should return both the speed setting and (if supported) the speed percent",  but from what I can tell, no further changes are needed.

I uploaded the fix to my github, so it should now be reflected in the Pull request.

Thanks!